### ISSUE TITLE
No Audio for 2 Android devices sharing same Room SessionID

### ISSUE THREAD

**Issue by aaronvargas** [EricssonResearch/openwebrtc-examples#46]
Overview:
Using the NativeCall app, the audio (and video) works correctly when connecting 1 Android device and a browser client at http://demo.openwebrtc.io:38080. Connecting 2 Android devices using NativeCall (and no browser client) results in shared video, but no audio.

Expected Behavior:
2 Android NativeCall apps should be able to join the same sessionID and share audio and video.

Actual Behavior:
2 Android apps in above scenario can only share video.

Environment: 
Browser - Chrome on mac (Yosemite)
Nexus 5 - 5.0.1
Nexus 7 (2013) - 5.0.2

Other Notes: 
I've connected 2 Android native clients (independent of the demo app and NativeCall) and got the video working. However audio didn't. As a sanity check I tried the outlined case above and found it didn't work in the demo app either.

When the audio IS (yes is) Working I see these errors in the log
02-07 10:36:11.098  16772-16960/com.ericsson.research.owr.examples.nativecall I/gst_log﹕ 0xa31536c0 ERROR   default /Users/aaron/openwebrtc/scripts/dependencies/gst-plugins-base/gst-libs/gst/audio/audio-info.c:267:gst_audio_info_from_caps: no layout given
02-07 10:36:11.198  16772-16792/com.ericsson.research.owr.examples.nativecall E/g_printerr﹕ Error in element audio-source.
02-07 10:36:11.198  16772-16792/com.ericsson.research.owr.examples.nativecall E/g_printerr﹕ Debugging info: /Users/aaron/openwebrtc/scripts/dependencies/gstreamer/libs/gst/base/gstbasesrc.c(2943): gst_base_src_loop (): /GstPipeline:local-audio-capture-source-bin-1/GstOpenSLESSrc:audio-source:
    streaming task paused, reason not-negotiated (-4)
02-07 10:36:11.834  16772-16792/com.ericsson.research.owr.examples.nativecall W/MediaController﹕ remote audio: com.ericsson.research.owr.RemoteMediaSource@33dfe211
02-07 10:36:11.966  16772-16988/com.ericsson.research.owr.examples.nativecall W/AudioTrack﹕ AUDIO_OUTPUT_FLAG_FAST denied by client
02-07 10:36:11.985      184-830/? D/audio_hw_primary﹕ out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=4
02-07 10:36:12.005      184-750/? D/audio_hw_primary﹕ select_devices: out_snd_device(5: headphones) in_snd_device(0: none)
02-07 10:36:12.005      184-750/? D/msm8974_platform﹕ platform_send_audio_calibration: sending audio calibration for snd_device(5) acdb_id(10)
02-07 10:36:12.005      184-750/? D/audio_hw_primary﹕ enable_snd_device: snd_device(5: headphones)
02-07 10:36:12.011      184-750/? D/audio_hw_primary﹕ enable_audio_route: apply and update mixer path: low-latency-playback

These errors are also present when audio is not working...

**Comment by stefanalund** [comment]
@Rugvip any thoughts?

**Comment by superdump** [comment]
Just to check: are you running the two instances of the all on two separate devices?

**Comment by aaronvargas** [comment]
Yes, I run one instance of NativeCall on a Nexus 5 and one on a Nexus 7. I then join the same sessionId (room) and initiate the call from either side. (there is often a delay at this point...) I then get 2-way video, but no audio. @superdump, Does it work for you?

**Comment by edmandiesamonte** [comment]
Any update on this? I'm also getting the same error. Tried the latest NativeCall sample app.
My device is Sony Xperia Z3 running on Android Lollipop 5.1. I used the demo website through Chrome on Mac. The video and audio are coming perfectly from Chrome on Mac to Android but video only if it's the other way around.

**Comment by Rugvip** [comment]
This issue is fixed by https://github.com/EricssonResearch/cerbero/commit/20a368b9a9c329a84b9cb7a6aea773b2692fd519 and https://github.com/EricssonResearch/openwebrtc/commit/6d72e0aceab36d185556dc987502026a8d936e72
If you fetch the latest cerbero master and rebuild, things should hopefully work.

**Comment by edmandiesamonte** [comment]
I tried it with the latest cerbero master with that same device, Sony Xperia Z3 running on Lollipop 5.1. I also updated the openwebrtc-examples. Result:

The gstreamer error no longer exists but the following log was still there:
```
04-15 14:57:43.547    3239-3239/com.ericsson.research.owr.examples.nativecall I/SimpleStreamSet﹕ audio stream mode set: SEND_RECEIVE
04-15 14:57:43.571    3239-5482/com.ericsson.research.owr.examples.nativecall W/AudioRecord﹕ AUDIO_INPUT_FLAG_FAST denied by client
04-15 14:57:43.839    3239-3239/com.ericsson.research.owr.examples.nativecall V/NativeCall﹕ candidate: {"sdpMLineIndex":0,"sdpMid":"audio","candidate":"candidate:3087517168 1 udp 2122260223 192.168.252.46 57730 typ host generation 0","candidateDescription":{"foundation":"3087517168","componentId":1,"transport":"UDP","priority":2122260223,"address":"192.168.252.46","port":57730,"type":"host"}}
04-15 14:57:43.890    3239-5502/com.ericsson.research.owr.examples.nativecall D/AudioTrack﹕ TrackOffload: AudioTrack Offload disabled by property, returning false
04-15 14:57:43.895    3239-5502/com.ericsson.research.owr.examples.nativecall W/AudioTrack﹕ AUDIO_OUTPUT_FLAG_FAST denied by client
```

**Comment by DmitrySkripunov** [comment]
Hello!
I have some issue with audio
to console write:
`W/AudioTrack: AUDIO_OUTPUT_FLAG_FAST denied by client` at all.... I don't know what to doing
I need any ideas, please)
